win/tty/termcap.c

```diff
@@ -327,13 +326,29 @@ tty_shutdown()
 #ifdef TERMLIB
 #ifdef TEXTCOLOR
     kill_hilite();
-#endif
+#endif /* TEXTCOLOR */
     if (dynamic_HIHE) {
         free((genericptr_t) nh_HI), nh_HI = (char *) 0;
         free((genericptr_t) nh_HE), nh_HE = (char *) 0;
         dynamic_HIHE = FALSE;
     }
-#endif
+#else /* TERMLIB */
+#ifndef TOS
+#if defined(ANSI_DEFAULT) && defined(TEXTCOLOR)
+    for (int i = 0; i < CLR_MAX / 2; i++)
+        if (i != CLR_BLACK) {
+            free(hilites[i | BRIGHT]);
+            if (i != CLR_GRAY)
+#ifdef MICRO
+                if (i != CLR_BLUE)
+#endif /* MICRO */
```

Comment:
I this section of code tied to a different section in this file? Why `MICRO` and `TOS`? If these a refer to a section of code where you literally deallocate what has been allocated, could you maybe put a small comment in here?

Reply:
This mirrors the logic at allocation time which starts here: https://github.com/facebookresearch/nle/blob/drop-ncurses/win/tty/termcap.c#L156

Basically, depending on these variables different entries of `hilites` are themselves `alloc`'d or just copies of other `alloc`'d pointers. (And we can't free multiple times.)

Reply:
Good point btw, I added a comment.